projects/storefrontlib/src/cms-components/configurator/commons/config-attribute-types/config-attribute-numeric-input-field/config-attribute-numeric-input-field.component.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  formatNumber,
+  getLocaleNumberSymbol,
+  NumberSymbol,
+} from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { AbstractControl, FormControl, ValidatorFn } from '@angular/forms';
+import { Configurator, LanguageService } from '@spartacus/core';
+import { ConfigFormUpdateEvent } from '../../config-form/config-form.event';
+import { ConfigUIKeyGeneratorService } from '../../service/config-ui-key-generator.service';
+
+@Component({
+  selector: 'cx-config-attribute-numeric-input-field',
+  templateUrl: './config-attribute-numeric-input-field.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ConfigAttributeNumericInputFieldComponent implements OnInit {
+  attributeInputForm: FormControl;
+  numericFormatPattern;
+  locale: string;
+
+  constructor(
+    public uiKeyGenerator: ConfigUIKeyGeneratorService,
+    public languageService: LanguageService
+  ) {}
+
+  @Input() attribute: Configurator.Attribute;
+  @Input() group: string;
+  @Input() ownerKey: string;
+
+  @Output() inputChange = new EventEmitter<ConfigFormUpdateEvent>();
+
+  mustDisplayValidationMessage(): boolean {
+    const wrongFormat: boolean =
+      (this.attributeInputForm.dirty || this.attributeInputForm.touched) &&
+      this.attributeInputForm.errors?.wrongFormat;
+
+    return wrongFormat;
+  }
+
+  determineCurrentLocale() {
+    //locales are available as languages in the commerce backend
+    this.languageService
+      .getActive()
+      .subscribe((language) => (this.locale = language))
+      .unsubscribe();
+  }
+
+  compilePatternForValidationMessage(
```

Comment:
Can we move business logic to a component-service so customers can easily extend it?